### ISSUE TITLE
"process stdout" test failure

### ISSUE THREAD

**Issue by jamessan** [luvit/luv#421]
While trying to update Debian's lua-luv package to the latest release, I'm running into a test failure with `53 process - process stdout`:

```
ok 52 process - test disable_stdio_inheritance
  { handle = uv_process_t: 0x562dcab718a0, pid = 27970 }
  "stdout"      {  }
Uncaught Error: ./tests/test-process.lua:29: assertion failed!
stack traceback:
        [C]: in function 'assert'
        ./tests/test-process.lua:29: in function 'fn'
        ./lib/tap.lua:48: in function <./lib/tap.lua:46>
        [C]: in function 'run'
        ./lib/tap.lua:55: in function <./lib/tap.lua:42>
        [C]: in function 'xpcall'
        ./lib/tap.lua:42: in function 'run'
        ./lib/tap.lua:136: in function 'tap'
        tests/run.lua:35: in main chunk
  "exit"        { signal = 0, code = 1 }
UNCLOSED        uv_pipe_t: 0x562dcab39550
  ./lib/tap.lua:70: 1 unclosed handle
  stack traceback:
        [C]: in function 'error'
        ./lib/tap.lua:70: in function <./lib/tap.lua:42>
        [C]: in function 'xpcall'
        ./lib/tap.lua:42: in function 'run'
        ./lib/tap.lua:136: in function 'tap'
        tests/run.lua:35: in main chunk
not ok 53 process - process stdout
  { handle = uv_process_t: 0x562dcab718a0, pid = 27971 }
  "exit"        uv_process_t: 0x562dcab718a0    { status = 0, signal = 2 }
```

This test works normally when simply running Lua, loading the module, and running the tests.  However, the Debian packaging also runs the tests with a small [shim app](https://sources.debian.org/src/dh-lua/25/test/5.1/app.c/) that directly links against the compiled Lua module and that's when the test fails.

```
libtool --silent --tag=CC --mode=link cc -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr//include/lua5.1 -I/usr/include/lua-compat53 -Wall -Wextra -Wl,--no-add-needed \
        -o 5.1-luv/app-dynamic -I . -I 5.1-luv/ \
        /usr/share/dh-lua/test/5.1/app.c 5.1-luv/liblua5.1-luv.la \
        -Wl,-z,relro -luv -lrt -lpthread -lnsl -ldl -L/usr//lib/x86_64-linux-gnu -llua5.1
```

The test worked with luv [1.9.1](https://buildd.debian.org/status/fetch.php?pkg=lua-luv&arch=amd64&ver=1.9.1-0-2%2Bb1&stamp=1564436825&raw=0), so something has regressed between then and 1.30.1-1.

**Comment by squeek502** [comment]
I would assume this is due to the `process stdout` test assuming `uv.exepath()` is a normal Lua runtime and passing `-e "io.write 'Hello World'"` to it. The shim app probably doesn't handle `-e` and so doesn't print `Hello World` causing the test to fail.

That assertion was added [in this commit](https://github.com/luvit/luv/commit/27a5b5f5b7c131cf02ee1d9fd8085a90bcad6e00#diff-7a1f5ee0d5ed37d49924a2c9929b7de6R29) so presumably the test *should* have been failing with 1.9.1 but it just wasn't actually checking the output back then.

Probably would be a good idea to make the test `uv.exepath()` independent if possible (i.e. run something that is guaranteed to print to stdout for each platform).

**Comment by squeek502** [comment]
Should be fixed by #423. Let us know if you run into any other issues.

**Comment by jamessan** [comment]
That did fix the issue.  Thanks for the quick response!
